File: codigo/soluciones/kat/hidden.py
pword, msg = input().split()
j = 0
passd = True
for i in range(len(pword)):
    if not passd:
        break
    found = False
    while j < len(msg) and pword[i] != msg[j]:
        for k in range(i + 1, len(pword)):
            if pword[k] != msg[j]:
                continue
            passd = False
            break
        j += 1
    if j < len(msg):
        found = True
        j += 1
    if not found:
        passd = False
print("PASS" if passd else "FAIL")

File: codigo/soluciones/kat/hidden.java
import java.io.*;
import java.util.*;
class Hidden {
    private void _main() {
        Scanner sc = new Scanner(new BufferedInputStream(System.in));
        String pword = sc.next(), msg = sc.next();
        int j = 0;
        boolean pass = true;
        for (int i = 0; i < pword.length(); ++i) {
            if (!pass) break;
            boolean found = false;
            while (j < msg.length() && pword.charAt(i) != msg.charAt(j)) {
                for (int k = i + 1; k < pword.length(); ++k) {
                    if (pword.charAt(k) == msg.charAt(j)) {
                        pass = false;
                    }
                }
                ++j;
            }
            if (j < msg.length()) {
                found = true;
                ++j;
            }
            if (!found) pass = false;
        }
        System.out.println(pass ? "PASS" : "FAIL");
        sc.close();
    }
    public static void main(String[] args) {
        (new Hidden())._main();
    }
}
